code/sha3/Hacl.Hash.SHA3.Scalar.fst

```diff
@@ -0,0 +1,176 @@
+module Hacl.Hash.SHA3.Scalar
+
+open FStar.HyperStack
+open FStar.HyperStack.ST
+open FStar.Mul
+
+open LowStar.Buffer
+
+open Lib.IntTypes
+open Lib.Buffer
+open Lib.ByteBuffer
+
+open Spec.Hash.Definitions
+open Hacl.Spec.SHA3.Vec
+open Hacl.Spec.SHA3.Vec.Common
+open Hacl.Impl.SHA3.Vec
+
+module S = Spec.SHA3
+module V = Hacl.Spec.SHA3.Vec
+module ST = FStar.HyperStack.ST
+module B = LowStar.Buffer
+module M = LowStar.Modifies
+
+#reset-options "--z3rlimit 50 --max_fuel 0 --max_ifuel 0 --using_facts_from '* -FStar.Seq'"
+
+val shake128:
+  output:buffer_t MUT uint8
+  -> outputByteLen:size_t{v outputByteLen == length output}
+  -> input:buffer_t MUT uint8
+  -> inputByteLen:size_t{v inputByteLen == length input}
+  -> Stack unit
+     (requires fun h ->
+       live h input /\ live h output /\ disjoint input output)
+     (ensures  fun h0 _ h1 ->
+       modifies (loc output) h0 h1 /\
+       as_seq h1 (output <: lbuffer uint8 outputByteLen) ==
+       S.shake128 (v inputByteLen) (as_seq h0 (input <: lbuffer uint8 inputByteLen)) (v outputByteLen))
+let shake128 output outputByteLen input inputByteLen =
+  admit();
```

Comment:
to facilitate the review of this PR, can you either:
- turn this admit into an assume, so that we understand what is missing
- add a comment describing what is missing?

Reply:
The implementation of these functions is not complete yet and it requires equivalent lemmas with original spec to prove post-condition and certain asserts to fulfill "multi-buffer" requirement of `keccak` function, I am working on equivalent lemmas and will update the functions once done.